hooks/useStakingData.ts

```diff
@@ -207,70 +207,67 @@ const useStakingData = () => {
 		args: [walletAddress ?? undefined, 0, 1000],
 		watch: false,
 		enabled: !!walletAddress,
 		select: (data) => data.filter((d) => d.escrowAmount.gt(0)),
-		onError(error) {
-			if (error) logError(error);
-		},
 	});
 
-	vestingSchedules &&
-		vestingSchedules.length > 0 &&
-		vestingSchedules.forEach((d: VestingEntry) => {
-			data.push({
-				id: Number(d.entryID),
-				date: moment(Number(d.endTime) * 1000).format('MM/DD/YY'),
-				time: formatTruncatedDuration(d.endTime - new Date().getTime() / 1000),
-				vestable: d.endTime * 1000 > Date.now() ? 0 : Number(d.escrowAmount / 1e18),
-				amount: Number(d.escrowAmount / 1e18),
-				fee: d.endTime * 1000 > Date.now() ? Number(d.escrowAmount / 1e18) : 0,
-				status: d.endTime * 1000 > Date.now() ? 'VESTING' : 'VESTED',
+	const escrowRows: EscrowRow[] = useMemo(() => {
+		if (vestingSchedules && vestingSchedules.length > 0) {
+			return vestingSchedules.map((d: VestingEntry) => {
+				return {
+					id: Number(d.entryID),
+					date: moment(Number(d.endTime) * 1000).format('MM/DD/YY'),
+					time: formatTruncatedDuration(d.endTime - new Date().getTime() / 1000),
+					vestable: d.endTime * 1000 > Date.now() ? 0 : Number(d.escrowAmount / 1e18),
+					amount: Number(d.escrowAmount / 1e18),
+					fee: d.endTime * 1000 > Date.now() ? Number(d.escrowAmount / 1e18) : 0,
+					status: d.endTime * 1000 > Date.now() ? 'VESTING' : 'VESTED',
+				};
 			});
-		});
+		}
+		return [];
+	}, [vestingSchedules]);
 
-	const contracts = data.map((d) => {
-		return {
-			...rewardEscrowContract,
-			functionName: 'getVestingEntryClaimable',
-			args: [walletAddress ?? undefined, d.id],
-			enabled: !!walletAddress,
-		};
-	});
+	const vestingEntries = useMemo(() => {
+		return escrowRows.map((d) => {
+			return {
+				...rewardEscrowContract,
+				functionName: 'getVestingEntryClaimable',
+				args: [walletAddress ?? undefined, d?.id],
+				enabled: !!walletAddress,
+			};
+		});
+	}, [escrowRows, rewardEscrowContract, walletAddress]);
 
-	const {
-		refetch: resetVestingClaimable,
-		data: vestingEntryClaimable,
-		isSuccess: vestingEntryClaimableIsSuccess,
-	} = useContractReads({
-		contracts,
+	const { refetch: resetVestingClaimable } = useContractReads({
+		contracts: vestingEntries,
 		watch: false,
-		enabled: !!walletAddress && contracts.length > 0,
+		enabled: !!walletAddress && vestingEntries.length > 0,
+		onSuccess(data) {
+			data.forEach((d, index) => {
+				escrowRows[index].vestable = Number(d?.quantity / 1e18) ?? 0;
+				escrowRows[index].fee = Number(d?.fee / 1e18) ?? 0;
+			});
+		},
 	});
 
-	vestingEntryClaimableIsSuccess &&
-		vestingEntryClaimable !== undefined &&
-		vestingEntryClaimable.forEach((d, index) => {
-			data[index].vestable = Number(d.quantity / 1e18);
-			data[index].fee = Number(d.fee / 1e18);
-		});
+	const totalVestable = useMemo(() => {
+		return escrowRows.reduce(
+			(acc, _current, index) => wei(acc).add(wei(escrowRows[index]?.vestable ?? 0)) ?? zeroBN,
```

Comment:
Instead of `(acc, _current, index) => wei(acc).add(wei(escrowRows[index]?.vestable ?? 0)) ?? zeroBN` could simplify with `(acc, row) => wei(acc).add(wei(row.vestable ?? 0)) ?? zeroBN` .